modules/Task/components/TaskDetails/TaskDetails.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState } from "react";
+import dayjs from "dayjs";
+import Swal from "sweetalert2";
+import { Badge, Box, Button, Card, Flex, Group, Text } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { IconEdit, IconTrash } from "@tabler/icons-react";
+
+import { EPriority, ETaskStatus } from "@/shared/typedefs/enums";
+import { TTask } from "@/shared/typedefs/types";
+import { useTasksContext } from "@/shared/utils/TasksProvider/TasksProvider";
+import EditTask from "@/modules/TodoAppPage/components/EditTask/EditTask";
+
+import { IProps } from "./TaskDetails.types";
+
+function TaskDetails({ task }: IProps) {
+  const [taskToEdit, setTaskToEdit] = useState<TTask | null>(null);
+  const [opened, { open, close }] = useDisclosure();
+  const { markTask, deleteTask } = useTasksContext();
+
+  const handleEditTask = (task: TTask) => {
+    setTaskToEdit(task);
+    open();
+  };
+
+  const handleDeleteTask = (task: TTask) => {
+    Swal.fire({
+      title: "Are you sure?",
+      icon: "warning",
+      showCancelButton: true,
+      confirmButtonColor: "#3085d6",
+      cancelButtonColor: "#d33",
+      confirmButtonText: "Yes, delete it!",
+    }).then((result) => {
+      if (result.isConfirmed) {
+        deleteTask(task);
+
+        Swal.fire({
+          title: "Deleted!",
+          text: "Your task has been deleted.",
+          icon: "success",
+        });
+      }
+    });
+  };
+
+  return (
+    <Card
+      key={task.id}
+      shadow="sm"
+      padding="lg"
+      radius="md"
+      style={{ width: "60%" }}
+      mx={"auto"}
+      mt={20}
+      withBorder
+    >
+      <Group justify="space-between" mt="sm" mb="xs">
+        <Text fw={500}>{task.title}</Text>
+        {task.status === ETaskStatus.ACTIVE && (
+          <Button
+            leftSection={<IconEdit />}
+            onClick={() => handleEditTask(task)}
+          >
+            Edit
+          </Button>
+        )}
+      </Group>
+
+      <Text size="md" c="dimmed" mb={4}>
+        {task.description}
+      </Text>
+
+      <Text size="md" c="dimmed" mb={4}>
+        Priority:{" "}
+        <Badge
+          color={
+            task.priority === EPriority.LOW
```

Comment:
Please move conditions to a separate function and then call it here, this look ugly. And please refrain from using nested ternary conditions.